### PR TITLE
#950: don't unhide duplicates of already opened tab

### PR DESCRIPTION
**Pull request by oksmelnik**
fix for https://github.com/mozilla/multi-account-containers/issues/950
Issue: If you open a tab, which was assigned to some containers and you have some hidden tabs from this container at the moment, all the hidden tabs will be opened again. 

Solution: check if the url of the hidden tab was already opened and if so skip it from re-opening to prevent duplicates.

### CODE REVIEW

File: src/js/background/messageHandler.js [hunk 1]
```diff
@@ -155,7 +155,24 @@ const messageHandler = {
...
+              throw e;
+            });
+
+            browser.tabs.onUpdated.removeListener(this.tabUpdateHandler);
+          }
+        };
+
+        // if it's a container tab wait for it to complete and
+        // unhide other tabs from this container
+        if (tab.cookieStoreId.startsWith("firefox-container")) {
```
Comment (groovecoder) [inline src/js/background/messageHandler.js]: Do we need this `if` check, or could we move this `addListener` into the above `if` that checks the `tab` is not `firefox-default` or `firefox-private` or a `moz-extension` url?

File: src/js/background/backgroundLogic.js [hunk 2]
```diff
@@ -113,11 +113,12 @@ const backgroundLogic = {
     return list.concat(containerState.hiddenTabs);
   },
 
-  async unhideContainer(cookieStoreId) {
+  async unhideContainer(cookieStoreId, toHide) {
```
Comment (groovecoder) [inline src/js/background/backgroundLogic.js]: Can we change this variable name from `toHide` to `alreadyShowingUrl` to make it more explicit?

File: src/js/background/messageHandler.js (hunk 1 above)
Comment (oksmelnik) [reply src/js/background/messageHandler.js]: yes, moving it makes perfect sense.  I'm not sure if 
`  firefox-default, firefox-private, nor our own confirm page loads `
would filter out all the non-container tabs, so added the additional check. Let me it you don't think it's necessary .

**Comment by groovecoder** [comment]
Resolved a conflict but this looks good now. Thanks!

**Pull request merged**